parsers/gdscript.c

```diff
@@ -0,0 +1,1343 @@
+/*
+*   Copyright (c) 2000-2003, Darren Hiebert
+*   Copyright (c) 2014-2016, Colomban Wendling <ban@herbesfolles.org>
+* 	Copyright (c) 2021, David Yang <davidyang6us@gmail.com>
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   This module contains functions for generating tags for GDScript language
+*   files. This module is derived from the Python module.
+*
+* 	GDScript language reference:
+* 	https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_basics.html
+*
+*/
+
+#include "general.h"  /* must always come first */
+
+#include <string.h>
+
+#include "entry.h"
+#include "nestlevel.h"
+#include "read.h"
+#include "parse.h"
+#include "vstring.h"
+#include "keyword.h"
+#include "routines.h"
+#include "debug.h"
+#include "xtag.h"
+#include "objpool.h"
+
+#define isIdentifierChar(c) \
+	(isalnum (c) || (c) == '_' || (c) >= 0x80)
+#define newToken() (objPoolGet (TokenPool))
+#define deleteToken(t) (objPoolPut (TokenPool, (t)))
+
+enum {
+	KEYWORD_class,
+	KEYWORD_func,
+	KEYWORD_extends,
+	KEYWORD_pass,
+	KEYWORD_return,
+	KEYWORD_lambda,
+	KEYWORD_variable,
+	KEYWORD_const,
+	KEYWORD_enum,
+};
+typedef int keywordId; /* to allow KEYWORD_NONE */
+
+typedef enum {
+	ACCESS_PRIVATE,
+	ACCESS_PROTECTED,
+	ACCESS_PUBLIC,
+	COUNT_ACCESS
+} accessType;
```

Comment:
Does GDScript the concept "access" ?

If yes, could you add a test case?

Reply:
GDScript borrowed python's logic of making anything with a name starting with an underscore_ protected, there's an _init() already in the test file. Are we expecting more information in the output for this?

Reply:
> Are we expecting more information in the output for this?

Yes. Let's the access information to expected.tags.
You can do it with putting
```
--fields=+{access}
```
to the args.ctags file.